### PR TITLE
feat: added docker build and release

### PR DESCRIPTION
**Pull request by ProvoK**
as per https://github.com/emad-elsaid/xlog/discussions/18 adding support for Docker.

Notes:
* I obviously cannot create secrets, this should be done by the maintainer (@emad-elsaid )
* **I could not test the GHA run**, given it's run just on release (tags). By the action docs, _should_ just work
* @emad-elsaid have you though about using [goreleaser](https://goreleaser.com/)? it can handle pretty easily binary/github/docker and OS packages releases with a simple config file

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+CONTRIBUTING.md
+LICENSE
+Makefile
+NOTICE
+README.md
+arm/
+powerpc/
+mips/
+build/docker/
+deploy/docker/
```
Comment (emad-elsaid) [inline .dockerignore:14]: we only have `README.md` and `LICENSE`, we don't have any of the rest. maybe we can exclude `/docs` directory too.

**Review by emad-elsaid — COMMENTED** [review]
Thanks for taking the time to work on this change, I everything looks good, I left one small change. 
How do you feel about using Github packages to publish the image instead of docker hub? it should remove one new actor from our dependencies

**Comment by emad-elsaid** [comment]
using goreleaser is a good idea, I haven't looked in depth tbh. I'll take some time to work on that

**Comment by ProvoK** [comment]
@emad-elsaid It's perfectly fine to use github packages, I'll update the PR as soon as I can including the dockerignore suggested changes!

**Comment by ProvoK** [comment]
I should've done, even the publishing to ghcr.io.

Again, not tested, you can play with this branch if you want. I've based my code from documentation 
(https://docs.github.com/en/actions/publishing-packages/publishing-docker-images#publishing-images-to-github-packages and https://github.com/docker/metadata-action#latest-tag)

**Comment by ProvoK** [comment]
I kept contributing/authors in the dockerignore given those files will eventually be present, I believe, in the repo.

lmk if you think they should be deleted nonetheless

File: build/docker/Dockerfile.alpine [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+WORKDIR /app
+COPY go.mod ./
+COPY go.sum ./
+RUN go mod download
+COPY ./ ./
+RUN go build -o xlog ./cmd/xlog
+
+FROM alpine as final
+COPY --from=builder /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/
```
Comment (emad-elsaid) [inline build/docker/Dockerfile.alpine:11]: why are we copying the certificates from the previous step here?

**Comment by emad-elsaid** [comment]
Thanks @ProvoK for taking the time to do this contribution, really appreciate it, I made some changes to simplify the structure and Dockerfile content. I hope that didn't break the release.yml

**Comment by emad-elsaid** [comment]
Nice, it worked 
https://github.com/emad-elsaid/xlog/pkgs/container/xlog

**Comment by ProvoK** [comment]
nice! sorry I didn't have time to update the PR yesterday, glad it worked 😄

**Pull request merged**